### ISSUE TITLE
Support custom ConsolePalette/ANSI colors

### ISSUE THREAD

**Issue by KatoStoelen** [cake-build/cake#2976]
Regarding #2966.

In build systems that have a build log with white background, the current palette is not... the prettiest of palettes. Would be great to have a way of customizing the colors. Could be by choosing between different "official" palettes, or by explicitly setting the different colors of the palette.

I suspect that you could achieve this using Cake.Frosting and a custom `IConsole` where you handle ANSI coloring yourself, but it would be nice utilize the built in ANSI coloring and do this without Cake.Frosting.

**Comment by pascalberger** [comment]
Related issue https://github.com/cake-build/cake/issues/2852

**Comment by augustoproiete** [comment]
> Could be by choosing between different "official" palettes, or by explicitly setting the different colors of the palette

Agreed. It would be nice if there were official palettes to choose from. Maybe start with two, one similar to the current one that looks good on dark backgrounds and an alternative one that looks good on light backgrounds.

Perhaps (as a separate issue) also add an extension point that can be used to set a different color palette from a module?

**Comment by secretMeedo** [comment]
I fixed a similar issue for my company. Please let me know if this issue is still open for contribution so I can investigate it further.

**Comment by abdelghani29** [comment]
Hello, can I try working on this issue?
